### PR TITLE
Parsing NodeProto and handling optional inputs of functions

### PR DESCRIPTION
**Pull request by gramalingam**
### Description
A few extensions/fixes:
(a) Expose node-level parsing in python, as it helps write node-level tests in python
(b) Fix an issue in function type/shape inference to support optional inputs.
(c) Add support for parsing/printing type-attributes

### CODE REVIEW

**Review by jcwchen — COMMENTED** [review]
nit: perhaps also add a Python test (parser_test.py) for parse_node in Python level?

File: onnx/parser.py [hunk 1]
```diff
@@ -51,3 +51,19 @@ def parse_function(function_text: str) -> onnx.FunctionProto:
         function_proto.ParseFromString(function_proto_str)
         return function_proto
     raise ParseError(msg)
+
+
+def parse_node(node_text: str) -> onnx.NodeProto:
+    """Parse a string to build a NodeProto.
+
+    Arguments:
+        node_text (string): formatted string
```
Comment (justinchuby) [inline onnx/parser.py:60]:
```suggestion
        node_text: formatted string
```

nit

**Comment by gramalingam** [comment]
> nit: perhaps also add a Python test (parser_test.py) for parse_node in Python level?

Done

**Comment by gramalingam** [comment]
Will close and re-open the PR (a new one), to check CODEOWNERS update works.

**Pull request closed without merging**